src/networkd.c

```diff
@@ -725,6 +725,24 @@ append_wpa_auth_conf(GString* s, const authentication_settings* auth)
     }
 }
 
+static void
+write_wpa_unit(net_definition* def, const char* rootdir)
+{
+    GString* s = g_string_new("[Unit]\n");
+    g_autofree char* path = g_strjoin(NULL, "/run/systemd/system/netplan-wpa-", def->id, ".service", NULL);
+    g_string_append_printf(s, "Description=WPA supplicant for netplan %s\n", def->id);
+    g_string_append(s, "DefaultDependencies=no\n");
+    g_string_append_printf(s, "After=sys-subsystem-net-devices-%s.device\n", def->id);
```

Comment:
it's worth noting here that we're not getting systemd's escaping here, and what we probably want is to shell out to `systemd-escape` to escape `def->id` as that's what these units will be named.

Reply:
Yes, let's make sure the names are escaped. I don't think we're likely to run into the issue often, but there might be the odd device that is handled special, like VLANs.

Reply:
Oh shit, forgot about that. Will do.